src/Core/Components/DataGrid/FluentDataGridRow.razor.cs

```diff
@@ -99,6 +100,58 @@ private async Task HandleOnCellFocusAsync(DataGridCellFocusEventArgs args)
         }
     }
 
-    Task IHandleEvent.HandleEventAsync(
-       EventCallbackWorkItem callback, object? arg) => callback.InvokeAsync(arg);
+    /// <summary />
+    internal async Task HandleOnRowClickAsync(string rowId)
+    {
+        if (Owner.Rows.TryGetValue(rowId, out var row))
+        {
+            if (Owner.Grid.OnRowClick.HasDelegate)
+            {
+                await Owner.Grid.OnRowClick.InvokeAsync(row);
+            }
+
+            if (row != null && row.RowType == DataGridRowType.Default)
+            {
+                foreach (var selColumn in Owner.Grid.SelectColumns)
+                {
+                    await selColumn.AddOrRemoveSelectedItemAsync(Item);
+                }
+            }
```

Comment:
If we go with single SelectColumn in FluntDataGrid, change to
```
if (row != null && row.RowType == DataGridRowType.Default && Owner.Grid.SelectColumn is not null)
{
    await Owner.Grid.SelectColumn.AddOrRemoveSelectedItemAsync(Item);
}
```

Reply:
As explained, I prefer to keep the "normal" operation of the columns. In most cases, the **foreach** will be performed on a item, but this isn't disturbing and shouldn't affect performance.